lxd/db/cluster.go

```diff
@@ -0,0 +1,470 @@
+	filter := ClusterGroupFilter{}
+	filter.Name = &name
+
+	objects, err := c.GetClusterGroups(filter)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to fetch ClusterGroup: %w", err)
+	}
+
+	switch len(objects) {
+	case 0:
+		return nil, ErrNoSuchObject
+	case 1:
+		return &objects[0], nil
+	default:
+		return nil, fmt.Errorf("More than one ClusterGroup matches")
+	}
+}
+
+// GetClusterGroupID return the ID of the ClusterGroup with the given key.
+// generator: ClusterGroup ID
+func (c *ClusterTx) GetClusterGroupID(name string) (int64, error) {
+	stmt := c.stmt(clusterGroupID)
+	rows, err := stmt.Query(name)
+	if err != nil {
+		return -1, fmt.Errorf("Failed to get ClusterGroup ID: %w", err)
+	}
+	defer rows.Close()
+
+	// Ensure we read one and only one row.
+	if !rows.Next() {
+		return -1, ErrNoSuchObject
+	}
+	var id int64
+	err = rows.Scan(&id)
+	if err != nil {
+		return -1, fmt.Errorf("Failed to scan ID: %w", err)
+	}
+	if rows.Next() {
+		return -1, fmt.Errorf("More than one row returned")
+	}
+	err = rows.Err()
+	if err != nil {
+		return -1, fmt.Errorf("Result set failure: %w", err)
+	}
+
+	return id, nil
+}
+
+// ClusterGroupExists checks if a ClusterGroup with the given key exists.
+// generator: ClusterGroup Exists
+func (c *ClusterTx) ClusterGroupExists(name string) (bool, error) {
+	_, err := c.GetClusterGroupID(name)
+	if err != nil {
+		if err == ErrNoSuchObject {
+			return false, nil
+		}
+		return false, err
+	}
+
+	return true, nil
+}
+
+// CreateClusterGroup adds a new ClusterGroup to the database.
+// generator: ClusterGroup Create
+func (c *ClusterTx) CreateClusterGroup(object ClusterGroup) (int64, error) {
+	// Check if a ClusterGroup with the same key exists.
+	exists, err := c.ClusterGroupExists(object.Name)
+	if err != nil {
+		return -1, fmt.Errorf("Failed to check for duplicates: %w", err)
+	}
+	if exists {
+		return -1, fmt.Errorf("This ClusterGroup already exists")
+	}
+
+	args := make([]interface{}, 2)
+
+	// Populate the statement arguments.
+	args[0] = object.Name
+	args[1] = object.Description
+
+	// Prepared statement to use.
+	stmt := c.stmt(clusterGroupCreate)
+
+	// Execute the statement.
+	result, err := stmt.Exec(args...)
+	if err != nil {
+		return -1, fmt.Errorf("Failed to create ClusterGroup: %w", err)
+	}
+
+	id, err := result.LastInsertId()
+	if err != nil {
+		return -1, fmt.Errorf("Failed to fetch ClusterGroup ID: %w", err)
+	}
+
+	// Insert nodes reference.
+	err = addNodesToClusterGroup(c.tx, int(id), object.Nodes)
+	if err != nil {
+		return -1, fmt.Errorf("Insert nodes for cluster group: %w", err)
+
+	}
+
+	return id, nil
+
+}
+
+// RenameClusterGroup renames the ClusterGroup matching the given key parameters.
+// generator: ClusterGroup Rename
+func (c *ClusterTx) RenameClusterGroup(name string, to string) error {
+	stmt := c.stmt(clusterGroupRename)
+	result, err := stmt.Exec(to, name)
+	if err != nil {
+		return fmt.Errorf("Rename ClusterGroup: %w", err)
```

Comment:
Should be "Failed to rename..." or "Failed renaming..." like in `CreateClusterGroup` above for consistency, and similar for the other errors in this function.